skops/card/_model_card.py

```diff
@@ -279,29 +206,135 @@ def select(self, key: str) -> Section:
         -------
         section : Section
             A dataclass containing all information relevant to the selected
-            section. Those are the title, the content, and subsections (in a
-            dict).
+            section. Those are the title, the content, subsections (in a dict),
+            and additional fields that depend on the type of section.
 
         Raises
         ------
         KeyError
             If the given section name was not found, a ``KeyError`` is raised.
+
         """
         section_names = split_subsection_names(key)
         # check that no section name is empty
         if not all(bool(name) for name in section_names):
             msg = f"Section name cannot be empty but got '{key}'"
             raise KeyError(msg)
 
-        section = self
+        section: Section = self
         for section_name in section_names:
             section = section.subsections[section_name]
         return section
 
+    def format(self) -> str:
+        """Format determines how the content of this section will be rendered."""
+        raise NotImplementedError
+
+    def __repr__(self) -> str:
+        """repr determines how the content of this section is shown in the
+        Card's repr"""
+        return self.content
+
+
+@dataclass(repr=False)
```

Comment:
Yes, it's a bit of a gotcha when you have `@dataclass` and inherit from a dataclass at the same time. But since the abstract class is removed now, this is no longer necessary.